core/src/storage2/collections/binary_heap/elements.rs

```diff
@@ -121,16 +138,14 @@ where
         assert!(a < self.len(), "a is out of bounds");
         assert!(b < self.len(), "b is out of bounds");
 
-        let old_a = self.get_mut(a).expect("index a must exist").take();
+        let child_info_a = self.get_child_mut(a).expect("index a must exist");
+        let a_opt = child_info_a.child.take();
 
-        let old_b = {
-            let b_opt = self.get_mut(b).expect("index b must exist");
-            let old_b = b_opt.take();
-            *b_opt = old_a;
-            old_b
-        };
+        let child_info_b = self.get_child_mut(b).expect("index b must exist");
+        let b_opt = core::mem::replace(child_info_b.child, a_opt);
 
-        *self.get_mut(a).expect("index a must exist") = old_b;
+        let child_info_a = self.get_child_mut(a).expect("index a must exist");
```

Comment:
Double query for the child which causes an unnecessary doubled query through the `BTreeMap` cache. Please try to avoid this especially in functions that are on the hot path of execution which `swap` most certainly is. There should be no need for the second `self.get_child_mut`.